### DISCUSSION TITLE
avalonia How to read gray images?  Do not use third-party libraries

### DISCUSSION THREAD

**Discussion by qfmee** [AvaloniaUI/Avalonia#17797] [Q&A]
How to read gray images?     
“  public Bitmap(string fileName) ”  This method reads images in 4-channel format, even if the original image is a single channel gray image

**Comment by thevortexcloud** [comment]
I don't know how to do this myself, but you can probably do it with Skia if Avalonia can't do it. Avalonia uses Skia for its backend on most platforms so you should already have a dependency for it without bringing in any extra library. `SkBitmap` seems to have several constructor overloads for changing how colours and what not are handled.

https://learn.microsoft.com/en-us/dotnet/api/skiasharp.skbitmap?view=skiasharp-2.88

**Reply by qfmee** [reply]
It doesn't seem to work, we have to reference the SkiaSharp library separately
